src/lib/auth.ts

```diff
@@ -5,17 +5,37 @@ import type { User } from "firebase/auth"
 import { app } from './firebase'
 
 export interface AuthState {
+  // auth.user will be either a firebase User object or null
   user: User | null
+  // auth.known is whether or not we have completed trying to login with firebase
+  // --i.e. if `true` then we "know" that we have a user or we "know" that we don't,
+  // but when the store is first initialized we don't "know" anything.
   known: boolean
 }
 
+// TODO: Why is this written as a function returning a store instead of export const auth={...}?
+// (in my testing it seems to work the other way, so I'm guessing just coding style?)
 const createAuth = () => {
+  
+  // Here we are creating a writable store, and destructuring it to obtain 
+  // the 'subscribe' and 'set' methods for later use in async functions.
+  // This is the same as `const user = writable({ user: null, known: false })`
+  // if we wanted to use user.set() in later code.
+  // here the user is null, 
```

Comment:
Again, this is simple re-stating what the code already defines. Familiarity with Svelte stores and their usage patterns is assumed.